### DISCUSSION TITLE
ARM Test Toolkit vs PS Rules for Azure

### DISCUSSION THREAD

**Discussion by vishalgoyal16444** [Azure/PSRule.Rules.Azure#1370] [Q&A] [answered]
We came across this today. And have used PS Rules for Azure in the past which also checks ARM templates against WAF rules from Azure. Wanted to ask if they both leverage the same rules library or cater to different checks ?

https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/test-toolkit

**Comment by BernieWhite** [comment, marked as answer]
> We came across this today. And have used PS Rules for Azure in the past which also checks ARM templates against WAF rules from Azure. Wanted to ask if they both leverage the same rules library or cater to different checks ?
> 
> https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/test-toolkit

The ARM ttk is a separate tool. There is work under way to bring these tools closer together. There is some overlap, however they have a different focus. The ttk focuses on recommendations for authoring templates instead of WAF.
